### ISSUE TITLE
How to understand this profiling report by py-spy?

### ISSUE THREAD

**Issue by lingvisa** [benfred/py-spy#371]
First, this is not an issue for py-spy, but I hope the author can help me understand what's happening with my profiling in py-spy. This is what happened:

Suddenly my program became extremely slow at one function call 'make_doc' and my program had only minor changes. I don't see why it became so slow. Inside the 'make_doc' there is Cython code which is hard to debug.

I tried to use py-spy to profile and generated 3 screenshots, but didn't get any clue on the slowness. It seems that it indicates the 'getmodule' method in inspect.py takes most time. Is that true? The most time spent on python interpreter code, not on my own code? Please help share your thoughts on profiling. I am on Mac Big Sur 11.2.3

<img width="1318" alt="114102093-94378d00-987b-11eb-8e98-f0424ba8ce50" src="https://user-images.githubusercontent.com/7905135/114242204-5b60ec00-993f-11eb-934d-abf80be8bb79.png">
<img width="1519" alt="114100861-c8aa4980-9879-11eb-84eb-482127b872a9" src="https://user-images.githubusercontent.com/7905135/114242231-63b92700-993f-11eb-9f19-84685e3f9942.png">
<img width="1324" alt="114099939-70bf1300-9878-11eb-99b2-2077ce54fe97" src="https://user-images.githubusercontent.com/7905135/114242244-6ae03500-993f-11eb-8213-dfbacfda5a83.png">

**Comment by lingvisa** [comment]
Two more notes:
1) I used 'sudo py-spy top -- python wbnlp_api.py --native' to generate the 2nd screenshot and in the make_doc function, there are cython code. But report didn't show how the function perform inside make_doc?

2) The GIL is almost 100%. Does that indicate the cause for the slowness?

**Comment by ThyrixYang** [comment]
I agree that a detailed instruction or tutorial to read the output is necessary for beginners.

**Comment by Jongy** [comment]
You can take a look at https://rbspy.github.io/using-flamegraphs/, it's a nice starting point for analyzing flamegraphs.

@ThyrixYang , "active" means time that the process/thread was running on the CPU; "GIL" means the time the thread was owning CPython's GIL. While owning the GIL you're also active, but it's not necessary the other way around: a thread can be active without owning the GIL (for example: a native extension that gets called from Python, releases the GIL and runs some code meanwhile)

**Comment by ThyrixYang** [comment]
@Jongy  Thanks for your kind reply!
